src/thinkspace/api/thinkspace-readiness-assurance/app/models/thinkspace/readiness_assurance/assessment.rb

```diff
@@ -29,14 +29,109 @@ def sync_assessments
         phase      = self.authable
         assignment = Thinkspace::Casespace::Assignment.find(phase.assignment_id)
 
-        if assignment.sync_rat
+        if assignment.sync_rat?
           assessments = Thinkspace::ReadinessAssurance::Assessment.where(authable: assignment.thinkspace_casespace_phases).without(self)
           assessments.each do |assessment|
             assessment.update_columns(questions: self.questions, answers: self.answers, settings: self.settings.deep_merge(assessment.settings))
           end
         end
       end
 
+      def sync(options)
+        phase      = self.authable
+        assignment = Thinkspace::Casespace::Assignment.find(phase.assignment_id)
+
+        scoring_keys  = ['correct', 'attempted', 'no_answer', 'incorrect_attempt']
+        question_keys = ['type', 'random', 'ifat', 'justification']
+
+        if assignment.sync_rat?
```

Comment:
Reworked by creating the `Thinkspace::ReadinessAssurance::Sync::Assessment` class, which is created by the `Thinkspace::ReadinessAssurance::Assessment` `sync` method. Currently the same `sync` method functionality is simply transplanted to the `Sync::Assessment`